### PR TITLE
DM-44275: Remove version number access from Apdb interface

### PR DESCRIPTION
**Pull request by andy-slac**
Regular clients of Apdb interface do not need to know about versions, all version-related issues are handled internally. Removing to avoid accidental use of those methods.

### CODE REVIEW

File: tests/test_apdbCassandra.py [hunk 1]
```diff
@@ -103,10 +103,6 @@ def tearDown(self) -> None:
         query = f"DROP KEYSPACE {self.keyspace}"
         self._run_query(query)
 
-    if TYPE_CHECKING:
-        # For mypy.
-        def make_instance(self, **kwargs: Any) -> ApdbConfig: ...
-
```
Comment (kfindeisen) [inline tests/test_apdbCassandra.py:109]: Not sure why this was removed. Should it be a separate commit?

File: python/lsst/dax/apdb/cassandra/apdbCassandra.py [hunk 2]
```diff
@@ -449,12 +449,14 @@ def _get_version(key: str, default: VersionTuple) -> VersionTuple:
 
     @classmethod
     def apdbImplementationVersion(cls) -> VersionTuple:
```
Comment (kfindeisen) [inline python/lsst/dax/apdb/cassandra/apdbCassandra.py:451]: While most people will use the `Apdb` interface instead of the implementation classes, should this be renamed to `_apdbImplementationVersion` just in case? As it is, it will still show up in [the public docs](https://pipelines.lsst.io/v/daily/py-api/lsst.dax.apdb.ApdbSql.html#lsst.dax.apdb.ApdbSql.apdbImplementationVersion).

**Review by kfindeisen — APPROVED** [review]
Just a few minor comments.

File: tests/test_apdbCassandra.py (hunk 1 above)
Comment (andy-slac) [reply tests/test_apdbCassandra.py:109]: It was a leftover for older refactoring, should have removed it then. I can split the commit.

File: python/lsst/dax/apdb/cassandra/apdbCassandra.py (hunk 2 above)
Comment (andy-slac) [reply python/lsst/dax/apdb/cassandra/apdbCassandra.py:451]: Well, subclasses can add public methods, I see no problem with that. Most people will see `Apdb` and should need to look at implementaion classes.
